packages/template-retail-react-app/app/main.jsx

```diff
@@ -12,3 +12,13 @@ const main = () => {
 }
 
 main()
+
+window.onload = () => {
+    window.STOREFRONT_PREVIEW = {}
+
+    if (window.STOREFRONT_PREVIEW) {
+        window.STOREFRONT_PREVIEW.getToken = () => {
+            return window.localStorage.getItem('token')
+        }
+    }
```

Comment:
```suggestion
    window.STOREFRONT_PREVIEW = {
       ...window.STOREFRONT_PREVIEW,
       getToken: () => window.localStorage.getItem('token')
     }
```
1. We don't want to replace the namespace if it already exists, just extend it.
2. Don't need to check if it exists immediately after we define it.